esp/bindings/http/platform/httpservice.cpp

```diff
@@ -987,17 +987,18 @@ EspHttpBinding* CEspHttpServer::getEspHttpBinding(EspAuthRequest& authReq)
         //is set to nullptr only if !authReq.isSoapPost.
         if (!authReq.isSoapPost && espHttpBinding && !espHttpBinding->isValidServiceName(*authReq.ctx, authReq.serviceName.str()))
```

Comment:
@mayx so I think what this change really does is bypass this check of "isValidServiceName()".  That is if that fails it sets espHttpBinding to nullptr, but then with your change that gets overridden again by the default binding below.  As if this check never happened?

That may be ok, but I'd like to understand why.  In this case, what is authReq.serviceName set to, and why is isValidServiceName() failing?  It should be calling EsdlBindingImpl ::isValidServiceName() which should look up based on the loaded ESDL definition / binding.. it should be fully dynamic, and have nothing to do with the ESDLbinding service name from esp.xml.

Reply:
My guess is it's comparing to the esp.xml ESP service name, when it should be using the dynamically loaded ESDL service name.

Reply:
@afishbeck This issue happens when the page first loads and the browser tries to display the default main page, so the service name is empty. This happens with all single ESP services, not just DESDL service. For this case the default binding definitely should be used. Even in the more general case when service name is not empty, if we're returning the default binding for multiple services, I don't think we should treat a single service differently by returning null. It's true that for a single service the check seems redundant as the single service is always being treated as the default binding. I kept the check to (1) be consistent with how it's done in the processRequest function and (2) in case we want to change the qualification of default binding. So I think the change is correct here in terms of functionality. We can probably simplify it by removing the check, and maybe we should do it for the processRequest function as well and possibly unify the code, that could be done with this issue, or create another issue.

Reply:
Ok, after discussing offline, leave as is.